Modules/mathmodule.c

```diff
@@ -824,37 +824,44 @@ m_log10(double x)
     }
 }
 
-
-/*[clinic input]
-math.gcd
-
-    x as a: object
-    y as b: object
-    /
-
-greatest common divisor of x and y
-[clinic start generated code]*/
-
 static PyObject *
-math_gcd_impl(PyObject *module, PyObject *a, PyObject *b)
-/*[clinic end generated code: output=7b2e0c151bd7a5d8 input=c2691e57fb2a98fa]*/
+math_gcd(PyObject *module, PyObject *const *args, Py_ssize_t n)
 {
-    PyObject *g;
-
-    a = PyNumber_Index(a);
-    if (a == NULL)
-        return NULL;
-    b = PyNumber_Index(b);
-    if (b == NULL) {
-        Py_DECREF(a);
-        return NULL;
+    PyObject *g, *item, *in;
+    g = PyLong_FromLong(0);
```

Comment:
All Python C-API calls need to be checked for their return value; that includes this one, and the `_PyLong_GCD` call below.